GVFS/GVFS.Common/ModifiedPathsDatabase.cs

```diff
@@ -52,6 +52,34 @@ public static bool TryLoadOrCreate(ITracer tracer, string dataDirectory, Physica
             return true;
         }
 
+        public void Compress(ITracer tracer)
+        {
+            int startingCount = this.modifiedPaths.Count;
+            using (ITracer activity = tracer.StartActivity("Compress ModifiedPaths", EventLevel.Informational))
+            {
+                foreach (var item in this.modifiedPaths)
+                {
+                    int pathSeparatorIndex = item.IndexOf('/');
+                    while (pathSeparatorIndex >= 0 && pathSeparatorIndex < item.Length - 1)
+                    {
+                        string folder = item.Substring(0, pathSeparatorIndex + 1);
+                        if (this.modifiedPaths.Contains(folder))

```

Comment:
Won't this end up deleting every folder from the collection? Every folder entry ends with a `/`, and so the `folder` variable will end up consuming the entire string, and then find a match, right?

Reply:
Hm, perhaps what's preventing that is the `item.Length - 1` up on line 63. That's super subtle, and easy to get wrong. Since we're doing a bunch of string parsing/creating here anyway, it seems far safer and not a big perf hit to just call `string.Split` on `/`, and then ensure that we only consider the path parts up to, but not including, the leaf part.

Reply:
I will test using split and see what the performance and memory use of that is.

Reply:
Memory usage was about the same and performance was
```
Current: {"startingCount":1839665,"EndCount":9528,"DurationMs":1409}
Using Split: {"startingCount":1839665,"EndCount":9528,"DurationMs":1990}
```

Since there isn't a huge difference in performance and using Split is more readable, I'll switch to use split.